File: app/src/main/java/com/tim/iot/local/LocalServer.java
package com.tim.iot.local;

import android.content.Context;
import android.content.SharedPreferences;
import com.tim.android.constant.AppConst;

/**
 * LocalServer
 *
 * @author Tell.Tim
 * @date 2019/12/27 14:55
 */
public class LocalServer implements ILocalServer {

    private Context context;

    public LocalServer(Context context) {
        this.context = context;
    }

    @Override
    public String getAccount() {
        SharedPreferences sharedPreferences =
                this.context.getSharedPreferences(AppConst.AUTH_SHARED_PREF, Context.MODE_PRIVATE);
        String authValue = sharedPreferences.getString(AppConst.AUTH_ACCOUNT_ITEM,
                AppConst.UN_AUTH_ACCOUNT_VALUE);
        if (authValue.equals(AppConst.UN_AUTH_ACCOUNT_VALUE)) {
            return "Empty account";
        } else {
            return authValue.substring(0, authValue.indexOf("&"));
        }
    }

    @Override
    public synchronized void saveAuthToLocal(String account) {
        SharedPreferences sharedPreferences =
                this.context.getSharedPreferences(AppConst.AUTH_SHARED_PREF, Context.MODE_PRIVATE);
        sharedPreferences.edit().putString(AppConst.AUTH_ACCOUNT_ITEM, account).apply();
    }

    @Override
    public synchronized void clearAuthorized() {
        //恢复默认的标记
        SharedPreferences sharedPreferences =
                this.context.getSharedPreferences(AppConst.AUTH_SHARED_PREF, Context.MODE_PRIVATE);
        sharedPreferences.edit()
                .putString(AppConst.AUTH_ACCOUNT_ITEM, AppConst.UN_AUTH_ACCOUNT_VALUE)
                .apply();
    }
}

File: app/src/main/java/com/tim/iot/device/local/LocalServer.kt
package com.tim.iot.device.local

import android.content.Context
import android.content.SharedPreferences
import com.tim.android.constant.AppConst

/**
 * LocalServer
 *
 * @author Tell.Tim
 * @date 2019/12/27 14:55
 */
class LocalServer(private val context: Context) : ILocalServer {

    override val account: String
        @Synchronized get() {
            val sharedPreferences =
                this.context.getSharedPreferences(AppConst.AUTH_SHARED_PREF, Context.MODE_PRIVATE)
            val authValue = sharedPreferences.getString(AppConst.AUTH_ACCOUNT_ITEM,
                    AppConst.UN_AUTH_ACCOUNT_VALUE)
            return if (authValue == AppConst.UN_AUTH_ACCOUNT_VALUE) {
                "Empty account"
            } else {
                authValue!!.substring(0, authValue.indexOf("&"))
            }
        }

    @Synchronized override fun saveAuthToLocal(accountInfo: String) {
        val sharedPreferences =
            this.context.getSharedPreferences(AppConst.AUTH_SHARED_PREF, Context.MODE_PRIVATE)
        sharedPreferences.edit().putString(AppConst.AUTH_ACCOUNT_ITEM, accountInfo).apply()
    }

    @Synchronized override fun clearAuthorized() {
        //恢复默认的标记
        val sharedPreferences =
            this.context.getSharedPreferences(AppConst.AUTH_SHARED_PREF, Context.MODE_PRIVATE)
        if (sharedPreferences.getString(AppConst.AUTH_ACCOUNT_ITEM,
                        AppConst.UN_AUTH_ACCOUNT_VALUE) != AppConst.UN_AUTH_ACCOUNT_VALUE
        ) {
            sharedPreferences.edit()
                    .putString(AppConst.AUTH_ACCOUNT_ITEM, AppConst.UN_AUTH_ACCOUNT_VALUE)
                    .apply()
        }
    }
}
